### PR TITLE
Script to download and restore annotated data

### PR DESCRIPTION
**Pull request by Golovneva**
**Patch description**
Updating ROSCOE with script to download and restore human-annotated data.
Updating links.

**Testing steps**
% python projects/roscoe/roscoe_data/restore_annotated.py
...
Saved SEMEVAL dataset in ./projects/roscoe/roscoe_data/generated/semevalcommonsense.json
Saved GSM8K dataset in ./projects/roscoe/roscoe_data/generated/gsm8k.json

Note: SEMEVAL dataset is not actually released yet. Commented out in the code.

### CODE REVIEW

File: projects/roscoe/roscoe_data/download_annotated.sh [hunk 1]
```diff
@@ -1,4 +1,38 @@
 #!/bin/bash
```
Comment (moyapchen) [inline projects/roscoe/roscoe_data/download_annotated.sh:1]: Ty for putting this together!

File: projects/roscoe/roscoe_data/restore_annotated.py [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+                quu2["answer"] = "yes"
+
+            quu1["premise"] = context + " " + q
+            quu2["premise"] = context + " " + q
+
+            if quu1["key"] in reasonings:
+                reasoning = reasonings[quu1["key"]]["reasoning"]
+                quu1["gpt-3"] = reasoning
+                # TODO: TMP, remove!
+                quu1["dataset"] = "semevalcommonsense_gpt3_expl"
```
Comment (moyapchen) [inline projects/roscoe/roscoe_data/restore_annotated.py:276]: This the right path?

File: projects/roscoe/roscoe_data/restore_annotated.py [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+def parse_gsm8k(fn, savefile):
+    with open(fn) as f:
+        lines = f.readlines()
+
+    structs = []
+    for line in lines:
+        data = json.loads(line.strip())
+        blob = {}
+        blob["premise"] = data["question"]
+        blob["hypothesis"] = (
```
Comment (moyapchen) [inline projects/roscoe/roscoe_data/restore_annotated.py:305]: lol, I wonder if we could get rid of this and the other code that strips this

File: projects/roscoe/roscoe_data/restore_annotated.py [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+            print(f"Saved DROP dataset in {save_file}")
+        elif dataset == 'esnli':
+            parse_esnli(input_file, model_output_reasoning, save_file)
+            print(f"Saved E-SNLI dataset in {save_file}")
+        elif dataset == 'cosmos':
+            # input_file = '/private/home/aslic/scorer/data/cosmosqa/data/valid.csv'
+            parse_cosmos(input_file, model_output_reasoning, save_file)
+            print(f"Saved COSMOSQA dataset in {save_file}")
+        elif dataset == 'semevalcommonsense':
+            # input_file = '/private/home/aslic/scorer/data/semevalcomsense/train_filter_maryam.xml'
```
Comment (moyapchen) [inline projects/roscoe/roscoe_data/restore_annotated.py:343]: "maryam" in the path

**Review by moyapchen — APPROVED** [review]
Looks reasonable for the most part, couple of strings

File: projects/roscoe/roscoe_data/restore_annotated.py (hunk 2 above)
Comment (Golovneva) [reply projects/roscoe/roscoe_data/restore_annotated.py:276]: lol, thank you for looking! this is not the path, that's key-value in the output JSON I added to completely match with your files, so I could easily run "diff". Will remove!

File: projects/roscoe/roscoe_data/restore_annotated.py (hunk 4 above)
Comment (Golovneva) [reply projects/roscoe/roscoe_data/restore_annotated.py]: thank you!

File: projects/roscoe/roscoe_data/download_annotated.sh [hunk 5]
```diff
@@ -1,4 +1,38 @@
...
-# Human annotated data: Pending data release approval
-# this file will contain a command to download annotated datasets into "roscoe_data/annotated" folder
-echo "Sorry, Pending data release approval"
+
+PATH_TO_DATA="./projects/roscoe/roscoe_data"
```
Comment (spencerp) [inline projects/roscoe/roscoe_data/download_annotated.sh:3]: nit; Usually data is downloaded to `ParlAI/data/{project}`, where the path to that `data/` folder is found in `opt['datapath']`. I think one reason is that `data/` is in the .gitignore (and there may be other assumptions coded in, in other places).

**Pull request merged**